src/style-spec/function/parse_expression.js

```diff
@@ -68,4 +87,21 @@ function parseExpression(expr: mixed, context: ParsingContext): ?Expression {
     }
 }
 
+const nonConstantExpressions = [ 'error', 'get', 'has', 'properties', 'id', 'geometry-type', 'zoom' ];
```

Comment:
This looks similar to `isFeatureConstant`/`isZoomConstant`. Can they share more code?

Reply:
👍 I almost did that, and then stopped myself because I thought I was prematurely abstracting.

Reply:
Oh, actually the code itself is a bit different. `isConstant` only looks one level deep (because that's sufficient for the way we're using it).  We could consolidate the base case code, though